server/src/main/java/org/elasticsearch/TransportVersions.java

```diff
@@ -193,6 +193,7 @@ static TransportVersion def(int id) {
     public static final TransportVersion ENRICH_ELASTICSEARCH_VERSION_REMOVED = def(8_560_00_0);
     public static final TransportVersion NODE_STATS_REQUEST_SIMPLIFIED = def(8_561_00_0);
 
+    public static final TransportVersion KNN_K_NUMCANDS_AS_OPTIONAL_PARAMS = def(8_993_00_0);
```

Comment:
Will update this to the proper version just before merging the PR

Reply:
Updated `KNN_K_NUMCANDS_AS_OPTIONAL_PARAMS` TransportVersion in 390aff2521de4c95f1a2e7260e7ed5cac6e4f1fa